Adopt changes from more maintained fork

CloCkWeRX opened the pull request:
Fix https://github.com/rospogrigio/localtuya/issues/1854
Fix https://github.com/rospogrigio/localtuya/issues/781

It took about an hour to merge, the only areas that need to be resolved as conflicts are versions, README and some translations.

Most relevant commits are the last 30.

This is @xZetsubou's branch plus a handful of other PRs for this repo.
Running locally for the past few days, a few devices going offline without logs; so TUYA protocol changes probably need to be double checked for merge conflicts.

1x failing test that is a merge conflict most likely.

xZetsubou committed:
Remove fan dps type
The order list should be always a list of speeds which will always will be sent as string, otherwise set rely on min/max and send as intgeer.

Lurker00 committed:
Call disconnect() only once for offlined sub-device (#344)

* Call disconnect() only once

* Typo error in a comment

ilankas committed:
fix: switch attrs unsupported operand type(s) for /: 'NoneType' and 'int' (#345)

* TypeError: unsupported operand type(s) for /: 'NoneType' and 'int'

* amended absed on review

Lurker00 committed:
remove unused variables in async_unload_entry

* Variable is not used anymore

* Variable was never used

mrpriv4te committed:
fix(cover): async_stop_after_timeout dead task (#355)

Co-authored-by: bkyrian <kyrian.bitoune@epitech.eu>

Lurker00 committed:
fix: actions order in async_setup_entry (#356)

xZetsubou committed:
Adds long term issues before stale.

francescolf committed:
import async_register_admin_service

Lurker00 committed:
refactor: shutdown entities on unexpected disconnect

* Unused parameter of "private" method

* Use main thread loop

* Don't reconnect to offline device much too often

* _shutdown_entities shall do its job on closing

* Delay _shutdown_entities with a sleep

* Git rid of call_later logic

Lurker00 committed:
fix: Don't update seqno from a delayed reply (#365)

* Don't update seqno from a delayed reply

* black format

---------

Co-authored-by: Bander <46300268+xZetsubou@users.noreply.github.com>

Lurker00 committed:
fix: async coroutine shall be awaited (#367)

Lurker00 committed:
fix: is_connected property (#368)

* Bug fix: self.transport shall not be None!

* Simplify condition

Co-authored-by: Bander <46300268+xZetsubou@users.noreply.github.com>

---------

Co-authored-by: Bander <46300268+xZetsubou@users.noreply.github.com>

Lurker00 committed:
perf: close connection improvements (#369)

* Eliminated unneeded self.heartbeater.cancel()

* Logical order of actions

* Direct call instead of await

* Wait for async tasks to finish on close

EamonnOL2 committed:
chore(typo): Zone Attributes (#371)

Fixing typo of "Sielnt" to "Silent"

felipecrs committed:
docs: Add remote entity to features in README (#381)

Lurker00 committed:
refactor: unload and shutdown (#363)

* Bug fix: self.transport shall not be None!

* Make TuyaProtocol.close waiting for activity stopped

* TuyaDevice.close() closes its sub-devices, if any.

* Different way to close devices on unload.

* Close devices on HA shutdown

* Fix: async coroutine shall be awaited

* Similar to #368

* Proposals by xZetsubou

xZetsubou committed:
chore: add alarm entities

xZetsubou committed:
fix: remove alarm deprecated constant #384

xZetsubou committed:
fix: negative values due to convert issues #354

xZetsubou committed:
perf(remote): add asyncio.lock to pervent duplicated commands, and ignore None dp codes

Lurker00 committed:
refactor: Explicit tasks cancel (#383)

* Eliminated _call_on_close, which was growing without a control

* Different cancellation order

* Clear the task on return

* Typo in the comment

* Clear after use

* Useless parameter

* Interruptable sleep

* Stop the previous _shutdown_entities and start a new one.

* Force debug logging of sleep cancel event

* Comment

* Fix mistaken edit

* Calling _shutdown_entities() in close() is not required.

* Early interrupt of _connect_subdevices

* Group cancel of async tasks

* Close detached sub-devices and faster close of sub-devices

* _sleep() actually is not required (anymore?)

* Don't remove sub-devices from its gateway's container

* Don't rely on disconnect(): _unsub_refresh in close()

* Don't count "absent" events as "offline" events

* try block around _async_reconnect loop body

* try block in _shutdown_entities

* More on cancel handling in _make_connection

* cleans up and missed expection 
* Merged abort connection w/ pending tasks.
* added cancled asyncio in pytuya connect.
* rename subdevice_state var/func

---------

Co-authored-by: Bander <46300268+xZetsubou@users.noreply.github.com>

xZetsubou committed:
feat: Add via_device for subdevice for homeassistant topology

xZetsubou committed:
feat: Add via_device for subdevice for homeassistant topology

xZetsubou committed:
Merge branch 'master' of https://github.com/xZetsubou/hass-localtuya

xZetsubou committed:
chore: removed test code

xZetsubou committed:
fix: blocking code and exporting messd dump #393

Lurker00 committed:
perf: transport write and close tasks order (#394)

* abort_connect() shall be called at the end: revert #383 abort interface close before sub-devices

* subdevice_state_updated calls

* Prevent out of order commands

* Better serialization of writes

* Check for _is_closing before going to sleep

xZetsubou committed:
fix: i/o blocking function of templates

* fix: i/o blocking function of templates

xZetsubou committed:
feat: disconnect device on disable

* disconnect on disable

* Entry will automatic reload after 30secs either way.

* cleans up

xZetsubou committed:
pref: Include sub device on disable disconnection (#400)

* Inlucde sub-devices on disable disconnection

* Fillter out closed subdevices from GW

* docstring

xZetsubou committed:
feat: add possibilty to add manual dps on reconfigure step

* add possibilty to add manual dps in reconfigure step

xZetsubou committed:
chore: rename private method to pbluc

xZetsubou committed:
feat: add lights mode sets #405

* added mode set option

xZetsubou committed:
bump version

xZetsubou committed:
chore: removed expectation for old HA versions

xZetsubou committed:
chore: removed tests codes

xZetsubou committed:
fix: IndexError - change fillter is entry check #410

xZetsubou committed:
bump ver

xZetsubou committed:
revert: qn category to heater instead of water heater #415

xZetsubou committed:
fix(auto_configure): device type case sensitivity #422

xZetsubou committed:
ci: added draft release

xZetsubou committed:
ci: revert header type

xZetsubou committed:
feat: add 0.01 precision for climates #437

xZetsubou committed:
fix: config_entry explicitly deprecated warnings #431

arnpere committed:
fix(fan): force integer for min/max speeds due to migration converting wrong type

After migration from upstream it seems fan min/max values change to str which is unexpected type for fans will lead to unavailable entitiy

Co-authored-by: Bander <46300268+xZetsubou@users.noreply.github.com>

geforcefan committed:
feat(light): Add Support for RGBW/RGBWW lights (#449)

* added white support to light entity

* added translations

* fixed a bug in has_config when checking for boolean selectors

* bumped version

* reverted some changes, removed white channel option in flow, now deriving single channel with existing logic and aligning with current architecture

* reverted some changes, removed white channel option in flow, now deriving single channel with existing logic and aligning with current architecture

* removed some templates which are pushed by accident

* changed supported_color_modes slightly for better readability

* changed supported modes 'brigthness' logic a bit

xZetsubou committed:
ci: fixes nichname author

xZetsubou committed:
feat: add ble device

This PR modifies the changes made in the original PR by @Lurker00

Co-authored-by: Lurker00 <kis@aha.ru>

xZetsubou committed:
fix(vacuum): update deprecated constant (#472)

xZetsubou committed:
fix(vacuum): Updated deprecated constant (#473)

* fix(vacuum): updated deprecated constant

* fix(vacuum): error state deprecated

xZetsubou committed:
fix(vaccum): use activity property  instead of deprecated state

* use activity instead of state

xZetsubou committed:
fixes(light): remove deprecated constant and improve value conversion (#477)

- Drops mireds properties and rely on kelvin.
- Drops deprecated constants.
- FIx map range method to give correct number.
- Use map range based on percent for brightness.
- Fix Temperature value are way off from Tuya APP.
- Set min default brightness to 10.

xZetsubou committed:
perf(cloud_api): Improve cloud_api speed by using aiohttp  (#471)

* refactor cloud_api to use async aiohttp

* Update LT to and remove hass object from cloud_api

slonopotamus committed:
ci: drop tox test (#475)

xZetsubou committed:
ci(draft_release): fix author name when it contains spaces (#483)

xZetsubou committed:
chore(auto_configure): adds EV charger
realted #479

xZetsubou committed:
feat(cover): handle boolean states (#485)

xZetsubou committed:
fix(cover): boolean current state

xZetsubou committed:
feat(climate): add support for string/number states (#484)

xZetsubou committed:
fix(config_flow): remove cloud pull label if dp exists locally

xZetsubou committed:
fix(climate): Preset mode always null (#488)

- Ensure the preset DP is the same HVAC mode DP before check.

- The condition would ensure that if preset and HVAC share same DP to not update preset mode if HVAC changed. this made because if preset contains more HVAC modes that aren't supported by HA would give user ability to change it otherwise state would be stuck to the latest one and would require to change it to another one then back.

xZetsubou committed:
feat(remote): Add support for rf commands and fix ENUM type IR commands

CloCkWeRX committed:
Merge branch 'master' of github.com:rospogrigio/localtuya

xZetsubou committed:
chore(auto_configure): adjust device class / network for qt category

xZetsubou committed:
fix(cover): hotfix for 2025.1 update and handle boolean actions state (#490)

* Auto configure current position for garage
* hotfix for latest update, commands not working.
* isupper check fix.
* handle is_closed for cover.
* handle states better

xZetsubou committed:
chore(auto_configure): set state class to total for energy device_class

related: #479

HepoH3 committed:
docs: fix broekn links

* Fix Get authorization data link
* fix configure_add_device link

xZetsubou committed:
ci(draft_release): use commit endpoint to pull username

omerfaruk-aran committed:
feat(language): Add support for Turkish language (#491)

* Create tr.json

* Reformatted `tr.json` file to use 2 spaces as default indentation.

xZetsubou committed:
fix(climate): add is_on property to support all on/off state types (#498)

gtgt committed:
perf(climate): Heuristic Action and fix heating (#468)

related #502

---------

Co-authored-by: Bander <46300268+xZetsubou@users.noreply.github.com>

xZetsubou committed:
chore: cleans up heuristic_action code

xZetsubou committed:
feat(cover): add support for close boolean state

related #489

xZetsubou committed:
chore(auto_configure): add Grakit dryer
related #501

xZetsubou committed:
chore(cover): revert state_action update from property

xZetsubou committed:
fix(light): Adjust brightness for encoded RGB bulbs #504

xZetsubou committed:
fix(light): map_range, reverse kelvin always return zero

xZetsubou committed:
fix(cloud_api): remove unnecessary calls (#513)

xZetsubou committed:
fix(climate): heuristic_action logic (#510)

* fix heuristic_action logic

* remove extra underscore

xZetsubou committed:
feat(diagnostics): includes discovered devices in downlaoded data (#514)

xZetsubou committed:
feat(sensors): handle raw data for such as "phase_x" (#511)

* Add support for base64 sensor data

* adjust phases code

* quick fix for unique id

* scale power sensor

* restore last state if it's raw.

xZetsubou committed:
ci: add code spell and refactor ci workflow (#515)

xZetsubou committed:
fix(climate): heuristic_action rely on temperature step instead of precision

related #502

xZetsubou committed:
fix(vacuum): fix pasue state in device has pause DP

xZetsubou committed:
ci: add tests workflow (#520)

xZetsubou committed:
ci(tests): update pyproject

eliandoran committed:
docs: update Tuya App Account point in cloud api (#444)

* docs: "Tuya App Account" -> "App Account" in cloud api

---------

Co-authored-by: xZetsubou <mrbanderx@gmail.com>

CloCkWeRX committed:
Update climate.py

CloCkWeRX committed:
Update climate.py

CloCkWeRX committed:
Update sensor.py

CloCkWeRX committed:
Merge commit '0f4c57083ca57b1161c136c25c59f1663a6fd3cd'

CloCkWeRX committed:
Merge commit 'e29456a52f1a1c588cfa27e187b198ebb8046436'

CloCkWeRX committed:
Merge commit 'e9c37e25ec3a33a03b938c8cc89ca735778c2477'

CloCkWeRX committed:
Merge commit 'd2a65d63db713e6567b29f87198783db45264525'

CloCkWeRX committed:
Merge commit 'ffc3a04f17429f5a7dad8150234df616fdfd3662'

CloCkWeRX committed:
Merge commit '8343003238446c19e1b1c9975c178f269b1381bd'

CloCkWeRX committed:
Merge commit 'bdd0b9e101176bf8702e66774483fc177c7af7ac'

CloCkWeRX committed:
Merge commit '977b8f1906c8310e07c85501831ed610a44c15c9'

CloCkWeRX committed:
Merge commit '04275d891266950b70b88797cbd7581c6e296f92'

CloCkWeRX committed:
Merge commit '021cd7001753349f2edbc792c9aff2e3456c2719'

CloCkWeRX committed:
Merge commit '7fa436511ecc80a72487686af60a163fa6281a3d'

CloCkWeRX committed:
Merge commit '31465864f0e07d98d41fb9218cbdbe5748bcbb12'

CloCkWeRX committed:
Merge commit '07780c32782ba29be32c04086c28d516f942edc9'

CloCkWeRX committed:
Merge commit '29b66f6d8b3415b640ff22df61a4a91c3def6d9c'

CloCkWeRX committed:
Merge commit 'fadaee87a60ae39ebab610c11a0de20b10fef81a'

CloCkWeRX committed:
Merge commit '3254eee84ffe6e5f8ba353469389a0a5f62506a0'

CloCkWeRX committed:
Merge commit 'cd476b42aecf742437acdcf588a19b01731b4c6f'

CloCkWeRX committed:
Merge commit 'edce3ccb9ad0a625d146fd95d0875fe9a4c00071'

CloCkWeRX committed:
Merge commit '56dbabec55e97a4297912ebade9e4d2798a4a7c4'

CloCkWeRX committed:
Merge commit 'acbba2e96521540f031f69a8ce62fbd3e1c7510b'

CloCkWeRX committed:
Merge commit 'b7ab0ca9e8df40014c749939264bd7fa511b9e76'

CloCkWeRX committed:
Merge commit '5bc7eed30f870ea2fb5636aa07f167f4c2fca456'

CloCkWeRX committed:
Merge commit 'ccb5c42a75104a6584c57528a2e403d6610536b4'

CloCkWeRX committed:
Merge commit 'b8d1a95fe635239b2b9deec9fc8d4f3e47d2cb42'

CloCkWeRX committed:
Merge commit 'eab53851dd24b939b3398074b0924b1c2d4f1137'

CloCkWeRX committed:
Merge commit '7111d73ed54e53a61bc9b3dde15a18602a5004a7'

CloCkWeRX committed:
Merge commit 'e5c9ad4637cfb1df41fd3a7e3bf4b8b5caa60603'

CloCkWeRX committed:
Merge commit '8a4a2e91d8f04ac08dabaa63d34b769537050362'

CloCkWeRX committed:
Merge commit 'ecbaa7101284e37c6b1f3ad6b3f75f3602f19fb6'

CloCkWeRX committed:
Merge commit '9fe52c8833411a9c56b5f7dafb174a76edc5c14c'

CloCkWeRX committed:
Merge commit '2f7248cb8c3d09ad961eab2532e900d90cbfffe3'

CloCkWeRX committed:
Merge commit '969b60bfd9c764072134b53a70090e93d935cfc0'

CloCkWeRX committed:
Merge commit '7a607a65087c9f88dc5f47d10678433d890afda3'

CloCkWeRX committed:
Merge commit '8f3ffe14b787c09e17f212d37f9461e2dd3fc376'

CloCkWeRX committed:
Merge commit 'c5f8bae1efa935bccf1ba531f50867731863f3e8'

CloCkWeRX committed:
Merge commit '2f733ce0dd99076b4936a7ce08642bca87ad4a93'

CloCkWeRX committed:
Merge commit '3a68cf6879636c2437d55eb4e93305f6b303369e'

CloCkWeRX committed:
Merge commit '504081ee6590589bdcb80993673f1ad291656633'

CloCkWeRX committed:
Merge commit '5e7c5f0d2df005785823cbac101185855b049b10'

CloCkWeRX committed:
Merge commit 'aafabc31a7a43ad86edc15fe35934d7ad5f1854e'

CloCkWeRX committed:
Merge commit '43d399bd4d69823c6fe78bb5f72b910714f564b1'

CloCkWeRX committed:
Merge commit '41fef89ff20ed67da7403b056f76dd25d4a4cd4d'

CloCkWeRX committed:
Merge commit 'a79735009d4feab3220f62e09917035d6f0c58ed'

CloCkWeRX committed:
Merge commit '8380743c10297dc39d07691ae417e6151b4d304d'

CloCkWeRX committed:
Merge commit 'f36568422258eb6e161dd3082a939b691e33bd91'

CloCkWeRX committed:
Merge commit 'e49b1fbbea32b39bb612f69b7988bd5a67a0283c'

CloCkWeRX committed:
Merge commit '3d0a22d005576c519ac4e67527d5efad7a428d8c'

CloCkWeRX committed:
Merge commit 'c82dc11aaaeb4b4ab3601fea4d7e88663718a520'

CloCkWeRX committed:
Merge commit 'a5e822f51556af5617472f7b10a26239f05de41d'

CloCkWeRX committed:
Merge commit '5a0b1c851474439207773a91d36d657f283b997c'

CloCkWeRX committed:
Merge commit 'c86724cc64ac456660dd86f04957d32ca2326c9e'

CloCkWeRX committed:
Merge commit 'dada3f4e60905ee50f24adae2545c084b3484fc5'

CloCkWeRX committed:
Merge commit '93cbb8c034215040fbc434448aec7edcad18374e'

CloCkWeRX committed:
Merge commit 'bb90383c3e78910fed044f1555a31b0d0844cdf6'

CloCkWeRX committed:
Merge commit '85a6c994718f641b6de1ed411e91d0d116dcb9d7'

CloCkWeRX committed:
Merge commit '5c264b45d4fe8e76843ebfc41c0c84b0e1258845'

CloCkWeRX committed:
Merge commit '14787e21986321f7a3d279dcd651bc62fae10624'

CloCkWeRX committed:
Merge commit '6a902e6e46c182539a069a8b80bce2870cefe076'

CloCkWeRX committed:
Merge commit '84bc95b8c26b3986fb0585c1504afe2d1208a72b'

CloCkWeRX committed:
Fix const

CloCkWeRX committed:
Fix include

CloCkWeRX committed:
Black style fixes

CloCkWeRX committed:
Typo

CloCkWeRX committed:
Sprelling

CloCkWeRX committed:
Sprelling

CloCkWeRX committed:
Fix merge issues

CloCkWeRX committed:
Remove undefined key

CloCkWeRX committed:
Remove dead code

CloCkWeRX committed:
Remove more dead code

CloCkWeRX committed:
Sync with upstream

CloCkWeRX committed:
Return correctly

CloCkWeRX committed:
Sync with upstream

CloCkWeRX commented on custom_components/localtuya/const.py:
```diff
@@ -85,15 +126,18 @@
-CONF_MIN_TEMP_DP = "min_temperature_dp"
-CONF_TEMP_MAX = "max_temperature_const"
-CONF_TEMP_MIN = "min_temperature_const"
+CONF_TEMP_MAX = "max_temp"
```
Note slight renaming.

xZetsubou committed:
fixes: as_dict DeviceConfig
related #528

xZetsubou commented:
Why do you think this is good idea, This is bad approach for an integration that being used by a thousands "upstream", The fork contains many changes related to core and __entry configuration__ the chances of breaking the old setups is really high especially for those who have a tons of devices with custom entities configurations, You may thought it working fine but the fork also breaks some features was existed in upstream since I thought it's not useful for me. It contains workarounds and dirty codes to support some type of devices, migration made long time ago wasn't targeted all configurations, days ago I found that in most cases fans will break as well and re-configure will be required.

I mentioned this many times before merging the fork with upstream won't be easy, I don't think it's worth the risks of breaking end-users setups.

CloCkWeRX committed:
Merge branch 'master' of https://github.com/xZetsubou/hass-localtuya

CloCkWeRX committed:
Merge climate shanges slightly

CloCkWeRX committed:
Merge climate shanges slightly

CloCkWeRX committed:
Adopt upstream

CloCkWeRX committed:
Adopt upstream

CloCkWeRX committed:
Adopt upsream

CloCkWeRX committed:
Remove sets

CloCkWeRX committed:
Remove sets

CloCkWeRX committed:
Remove sets

xZetsubou committed:
chore: subdevice shall not be added as "via_device" if it's gateway

phrak commented:
Better to abandon the upstream and deploy the fork.

CloCkWeRX commented:
I've stepped through the hundreds of changes. Is every merge conflict decision I have made correct? Probably not.
But there's zero effort to integrate bug fixes or PRs here. So right now, everyone using this integration is stuffed.
This has a small chance of making them not stuffed.

xZetsubou commented:
Resolving the merge conflicts is a thing and handling integration migration is a different thing. And the changes on your fork is accepting both changes of upstream/fork and merge them together which it doesn't makes sense.

xZetsubou committed:
ci(draft_release): fix special characters on commits (#534)

xZetsubou committed:
docs: update template example (#535)

xZetsubou committed:
docs: fix usage broken link in overview (#537)

xZetsubou committed:
ci: update codespell 2.4.1 (#538)

xZetsubou committed:
fix(climate): convert current temperature when unit is celsius/fahrenheit

xZetsubou committed:
cleanup some variables and methods names

xZetsubou committed:
feat(binary_sensor): add interval reset timer to set state off (#548)

xZetsubou committed:
feat(remote): Add sending head/key buttons commands (#550)

* add head and key parser and refactor some codes

* add docstrings

* feat: add remote services to handle manual adding of codes

* Remove the extra zero at start of the key

xZetsubou committed:
fix(climate): Update instance variable "hvac_action" when heuristic_action enabled (#556)

xZetsubou committed:
Revert "fix(climate): convert current temperature when unit is celsius/fahren…" (#557)

This reverts commit d757e4a665b143f9a5606362a89b5c793ea36cf1.

xZetsubou committed:
ci: ignore colon on prefix (#558)

felipecrs committed:
chore: fix typos and improve some sentences in README (#559)

* Fix typos and improve some sentences in README

* Update README.md

Co-authored-by: Bander <46300268+xZetsubou@users.noreply.github.com>

* Update README.md

Co-authored-by: Bander <46300268+xZetsubou@users.noreply.github.com>

* Update README.md

Co-authored-by: Bander <46300268+xZetsubou@users.noreply.github.com>

* Update README.md

---------

Co-authored-by: Bander <46300268+xZetsubou@users.noreply.github.com>

Lurker00 committed:
perf(core): fixes for low-performance HA servers (#555)

* Adjust the sleep time for true periodic events

* Avoid empty offline/online state parsing

* Increased timeout for connect. Named timeout values.

* Named timeout values.

* Apply suggestions from code review

Co-authored-by: Bander <46300268+xZetsubou@users.noreply.github.com>

* Apply suggestions from code review

Co-authored-by: Bander <46300268+xZetsubou@users.noreply.github.com>

* Black codestyle complain

* Monotonic timers for intervals

* Redundant task

* Faster is_sleep

* tabs -> spaces :(

* Apply suggestions from code review

Co-authored-by: Bander <46300268+xZetsubou@users.noreply.github.com>

* Eliminated now useless on_connected

---------

Co-authored-by: Bander <46300268+xZetsubou@users.noreply.github.com>

xZetsubou committed:
perf(cloud_api): Implement a shared aiohttp class session for managing requests (#554)

Lurker00 committed:
chore: clean up changes from PR #555 (#560).

xZetsubou committed:
fix(binary_sensor): reset state timer (#563)

Due to the logic of "_update_handler" if new status updated from device but the status aren't changed from cached one entity "status_updated" won't be called, this adds workaround to override the old cached status.

xZetsubou committed:
breaking_change(events): removed/rename events and improve events handler (#564)

xZetsubou committed:
docs: update events and services and fixes all links (#565)

* fix links

* update events.

* add remote services

* danger -> warning

xZetsubou committed:
docs: fix typos

xZetsubou committed:
feat(climate): Add swings support (#566)

* add vertical and horizontal swings

xZetsubou committed:
ci(tests): add climate swings (#571)

xZetsubou committed:
auto_configure: add more entities to cwwsq (#573)

devbis committed:
fix: correct closing devices on shutdown (#574)

CloCkWeRX committed:
Merge branch 'master' of https://github.com/xZetsubou/hass-localtuya

CloCkWeRX committed:
Fix merge conflicts

CloCkWeRX committed:
Fix merge conflicts

CloCkWeRX committed:
Fix merge conflicts

CloCkWeRX committed:
Fix merge conflicts; dropping exception handling

CloCkWeRX committed:
Fix tests

CloCkWeRX committed:
Temporarily remove accessor for swing mode

CloCkWeRX committed:
Temporarily remove accessor for swing mode

CloCkWeRX committed:
chore: trailing whitespace in en.json (#576)

Accidentally added in the fan swing mode PR

xZetsubou committed:
refactor: Improve handling of ObjectSelector configs (#569)

* Implement class to handle objectselector mapping

* fix climate test

* use `DictSelector` with alaram_control_panel

* use `DictSelector` with humidifier

* fix `DictSelector` with humidifier

* fix climate object has no attribute

* use `DictSelector` with select

* use `DictSelector` with water_heater

* avoid attr rename by python

* use `DictSelector` with lights

* add more docked status

* Clarify DictSelector docstrings and fix __repr__

xZetsubou committed:
fix(lights): `argument of type` error

xZetsubou committed:
fix(humidifier): local variable modes error (#591)

xZetsubou committed:
ci(test): fix humidifier tests introduced in 1b0e8c2c3066be78c67b92aa21b4d7dc975528a1

Lurker00 committed:
perf: better handling of absent subdevices (#584)

xZetsubou committed:
fix(entity): device classes not configured (#597)

xZetsubou committed:
auto_configure: added msp category

related #599
I added what looked obv for me, there still more codes not sure what it do

Lurker00 committed:
chore: fix typo and avoid unneeded operations (#605)

CloCkWeRX committed:
Merge branch 'master' of https://github.com/xZetsubou/hass-localtuya

xZetsubou committed:
revert removed categories

xZetsubou committed:
fix(cover): stuck on closed for no position mode (#625)

JohnnyPr1nce committed:
feature(language): Add support for 简体中文 (Simplified Chinese) (#627)

xZetsubou committed:
added entities to 'cwysj' category
related #616

xZetsubou committed:
added 'qxj' category 'Weather Station'
related #615

xZetsubou committed:
Add more entities to "msp" category
related #623

CloCkWeRX committed:
Merge branch 'master' of https://github.com/xZetsubou/hass-localtuya

xZetsubou committed:
pump version

CloCkWeRX committed:
Merge remote-tracking branch 'xZetsubou/master'

xZetsubou committed:
add funding

xZetsubou committed:
fix(alarm_control_panel): use alarm_state instead of state

xZetsubou committed:
ci(pytest): update alarm pytest to use alarm_state

CloCkWeRX committed:
ci: add a basic dev container and contributing doc (#646)

* Add basic devcontainer

* Add contributing doc

* Update CONTRIBUTING.md

xZetsubou committed:
fix(sensors): handle 0 values for subsensors (#653)

* Handle 0 values for sub sensors

* none as default value for getattr

* add qccdz category
related: 658

xZetsubou committed:
fix(lights): fix brightness number issue on some cases (#670)

xZetsubou committed:
ci(draft_release): add version into the release conecnt body

xZetsubou committed:
ci(draft_release): add version icon

xZetsubou committed:
perf: added info logs in make connection step

xZetsubou committed:
chore(config_flow): re-position submit message
related to PR:
https://github.com/xZetsubou/hass-localtuya/commit/81cc6301e511616fb819082ed072d4515bb6c1ef

smarthomeblack committed:
feat(languages): Add support for Vietnamese

Co-authored-by: truongblack <truongbber@gmail.com>

xZetsubou committed:
added entities for gcj category
related #680

xZetsubou committed:
feat(cloud): add Singapore Data Center
related #676

xZetsubou committed:
refactor(core): clean up and improve errs handler

xZetsubou committed:
perf(binary_sensor): remove possible on states, multi on value can be set

xZetsubou committed:
perf(binary_sensor): remove possible on states, multi on value can be set

related: #674

xZetsubou committed:
bump version

xZetsubou committed:
chore(core): cleanup debug messages

xZetsubou committed:
fix(core): decodeerror import

arielgr committed:
feat(remote): add support for sending raw codes (#690)

* Adding support for sending raw IR codes when device is not specified, in order to enable other components like SmartIR to send codes they store on their end.

* Addressing comments to keep device mandatory
and add special "raw_b64" device type for raw commands.

xZetsubou committed:
fix(fan): handle undefined speed value if not exists ordered list

related: #699

xZetsubou committed:
chore(core): typo check causing an false logs

CloCkWeRX committed:
Merge remote-tracking branch 'xZetsubou/master'

xZetsubou committed:
perf(config_flow): Allow empty inputs for optional fields (#719)

xZetsubou committed:
perf: Ignore fire events if no listeners (#712)

* Check and ignore fire event if no listeners

* remove debugging log

xZetsubou committed:
refactor(vacuum): Remove battery attrbiute (#720)

* Remove battery attr from vacuum

* remove battery test

CloCkWeRX committed:
Merge branch 'master' of https://github.com/xZetsubou/hass-localtuya
